utilities/include/alps/numeric/tensors/tensor_base.hpp

```diff
@@ -406,6 +406,14 @@ namespace alps {
         /// sizes for each dimension
         const std::array < size_t, Dim > &shape() const { return shape_; };
 
+        /// reshape with index list
+        template<typename Ind, typename ...Inds>
+        typename std::enable_if<std::numeric_limits<Ind>::is_integer>::type reshape(Ind ind, Inds...inds) {
```

Comment:
You can check all index types using the following metafunction:
```c++
#include <iostream>
#include <type_traits>

template<typename I0, typename... I> struct are_all_integers :
    std::integral_constant<bool, std::is_integral<I0>::value && are_all_integers<I...>::value>
{};
template<typename I0> struct are_all_integers<I0> : std::is_integral<I0> {};

int main()
{   
    std::cout << std::boolalpha << are_all_integers<int>::value << std::endl;
    std::cout << std::boolalpha << are_all_integers<int,char,unsigned long>::value << std::endl;
    std::cout << std::boolalpha << are_all_integers<int,double,unsigned long>::value << std::endl;
    
    return 0;
}
```
```
true
true
false
```

Reply:
There is also [`std::conjunction`](http://en.cppreference.com/w/cpp/types/conjunction) which we could provided in utilities.